### PR TITLE
Adding data factory CRUD scenario test

### PR DESCRIPTION
**Pull request by bgold09**
## Description

Adding data factory CRUD scenario test
  * This is the first batch of scenario tests for the ADF V2 SDK.
  * Partially addresses #3726; more tests need to be added
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by bgold09** [comment]
@shahabhijeet can you take a look at this? This is the first cut of scenario tests for the ADF V2 SDK. I wanted to see if the approach looked correct before continuing on with adding more. Thanks!

**Comment by hvermis** [comment]
Did you already add the project file for scenario tests? I don't see it in PR

**Review by hvermis — CHANGES_REQUESTED** [review]
:clock1:

**Review by hvermis — APPROVED** [review]
:shipit:

File: src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        [Fact]
+        [Trait(TraitName.TestType, TestType.Scenario)]
+        public void DataFactoryCrud()
+        {
+            var expectedFactory = new Factory(location: FactoryLocation);
+
+            Action<DataFactoryManagementClient> action = (client) =>
+            {
+                // create
```
Comment (hvermis) [inline src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs]: >// create [](start = 16, length = 9)

Minor comment - not a fan of comments in the code, it is self explanatory

Comment (bgold09) [reply src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs]: removed; these were originally comments for myself as I was creating the test

File: src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using System;
+using System.Linq;
+using Xunit;
+
+namespace DataFactory.Tests.ScenarioTests
+{
+    public class DataFactoryScenarioTests : ScenarioTestBase<DataFactoryScenarioTests>
+    {
+        [Fact]
+        [Trait(TraitName.TestType, TestType.Scenario)]
```
Comment (shahabhijeet) [inline src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs]: @bgold09 just FYI, we have not started using traits in SDK tests yet. All tests are executed as of now. But if you would like to categorize tests, go ahead do it. But please be aware the categories can be different when we introduce categories for the entire repo.

**Review by shahabhijeet — APPROVED** [review]

File: src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs (hunk 2 above)
Comment (bgold09) [reply src/SDKs/DataFactory/DataFactory.Tests/ScenarioTests/DataFactoryScenarioTests.cs]: sounds good, we were using these only for organizational purposes within our own test project

**Comment by bgold09** [comment]
@shahabhijeet is this enough coverage of scenario tests to unblock new code changes being merged? We plan to add more to get proper coverage in the coming weeks, but I was interested in knowing if this is acceptable for the time being. New features (like in PR #3757) should come in with new coverage and we will still need to go back and add tests for things that are already in the repo now

**Review by shahabhijeet — COMMENTED** [review]
See FYI in comments

**Pull request merged**